com.unity.shadergraph/Editor/GraphUI/GraphElements/ModelUI/GraphDataNode.cs

```diff
@@ -55,7 +37,29 @@ protected override void BuildContextualMenu(ContextualMenuPopulateEvent evt)
                 GraphView.Dispatch(new ChangePreviewExpandedCommand(false, new [] {m_GraphDataNodeModel}));
             });
 
-            // TODO: Add preview mode 2D/3D change submenu options to menu
+            evt.menu.AppendAction("Copy Shader", action =>
+            {
+                Debug.LogWarning("UNIMPLEMENTED in GraphDataNode");
```

Comment:
These are great, will serve as a good reminder for when we have preview stuff up and running 🤩